### ISSUE TITLE
Remove use of types module constants in twisted/mail

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8453]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8453|
|Type|enhancement|
|Created|2016-06-12 04:37:08Z|
|Branch|https://github.com/twisted/twisted/tree/8453-rodrigc-types|

The *types* module constants have been removed in Python 3.
For example using *types.TupleType* in Python 3 will give an *AttributeError*

These constants should be removed and replaced with the primitive type instead.  See:

http://www.diveintopython3.net/porting-code-to-python-3-with-2to3.html#types

<details><summary>Searchable metadata</summary>

```
trac-id__8453 8453
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8453_rodrigc_types 8453-rodrigc-types
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1465706228363653 1465706228363653
changetime__1468028949268917 1468028949268917
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/219

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Thanks for the updated. They look good with the exception of

`types.StringTypes -> str` conversion. I think it should be converted to `(str, unicode)`

Now I can see that the tests are green, so it looks like we have a hole in the tests.

Please consider my comment, and in case  we also need (unicode) in the list, please write a new test to cover unicode coverage on py2.

Please check my comment and resumit.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc commented|
|-|-|

Patch updated

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)|@rodrigc removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Hi it looks like we do need Unicode in that list, so we have a hole in the tests which is similar to missing coverage.

100% coverage does not only mean 100% line coverage.

Please add a tests for collapseNestedLists when using unicode items.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_8"></a>|@rodrigc removed owner|
|-|-|

Test updated

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban commented|
|-|-|

Changes look good. Thanks!

Please merge.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_10"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset 8fe97b3f224ebb5038f9f8873d48dd38b422c64e
```
#!CommitTicketReference repository="" revision="8fe97b3f224ebb5038f9f8873d48dd38b422c64e"
Merge pull request #219 from twisted/8453-rodrigc-types

Author: rodrigc
Reviewer: adiroiban
Fixes: #8453

Replace use of types constants with primitive types in twisted/mail
```
